src/targets/objc/native.js

```diff
@@ -1,57 +1,76 @@
 'use strict';
 
 var util = require('util');
+var objcHelpers = require('./helpers');
 
 module.exports = function (source, options) {
   var opts = util._extend({
-    timeout: '10'
+    timeout: '10',
+    indent: true
   }, options);
 
   var code = [];
 
-  // Dependencies
   code.push('#import <Foundation/Foundation.h>');
+
+  if (Object.keys(source.allHeaders).length) {
+    code.push(null);
+    code.push(objcHelpers.nsDictionaryBuilder('headers', source.allHeaders, opts.indent));
+  }
+
+  var parameters;
+
+  if (source.postData.mimeType && (source.postData.text || source.postData.params)) {
+    switch (source.postData.mimeType) {
+      case 'application/x-www-form-urlencoded':
+        parameters = source.postData.params;
+
+        code.push(null);
+        code.push('NSMutableData *postData = [[NSMutableData alloc] initWithData:[@"' + parameters[0].name + '=' + parameters[0].value + '" dataUsingEncoding:NSUTF8StringEncoding]];');
+
+        for (var i = 1, len = parameters.length; i < len; i++) {
+          code.push('[postData appendData:[@"&' + parameters[i].name + '=' + parameters[i].value + '" dataUsingEncoding:NSUTF8StringEncoding]];');
+        }
+        break;
+
+      case 'application/json':
+        try {
+          parameters = JSON.parse(source.postData.text);
+          code.push(objcHelpers.nsDictionaryBuilder('parameters', parameters, opts.indent));
+          code.push(null);
+          code.push('NSData *postData = [NSJSONSerialization dataWithJSONObject:parameters options:0 error:nil];')
+        } catch (e) {}
+        break;
+
+      case 'multipart/form-data':
```

Comment:
1. text used to not be set in that case
2. I used that option because it's simpler for developers to edit/remove a parameter once pasted. So really not sure.